app/models/report.rb

```diff
@@ -14,4 +23,59 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def link_detect_and_save
+    transaction_successful = false
+
+    ActiveRecord::Base.transaction do
+      raise ActiveRecord::Rollback unless save && linked_reports_save
+
+      transaction_successful = true
+    end
+
+    transaction_successful
```

Comment:
```suggestion
    ActiveRecord::Base.transaction do
      success = save && linked_reports_save
      raise ActiveRecord::Rollback unless success
      success
    end
```

ブロックの戻り値がそのままtransactionメソッドの戻り値になるのでこう書くこともできます

Reply:
そっちの方がきれいな感じがしますね！